files/image_config/interfaces/interfaces.j2

```diff
@@ -34,6 +34,41 @@ iface lo {{ 'inet' if prefix | ipv4 else 'inet6' }} static
 
 # The management network interface
 auto eth0
+{% if (ZTP_DHCP_DISABLED is not defined) and (ZTP is defined) and (ZTP['mode'] is defined and ZTP['mode']['profile'] == 'active') %}
+
+
+# ZTP out-of-band interface
+allow-hotplug eth0
+{% if ZTP['mode']['ipv4'] == 'true' %}
+iface eth0 inet dhcp
+{% endif %}
+{% if ZTP['mode']['ipv6'] == 'true' %}
+iface eth0 inet6 dhcp
+    up sysctl net.ipv6.conf.eth0.accept_ra=1
+    down sysctl net.ipv6.conf.eth0.accept_ra=0
+{% endif %}
+
+{% if ZTP['mode']['inband'] == 'true' %}
+{% for port in PORT %}
+{% set var = {'port_exists': True if port | is_intf_exists else False } %}
```

Comment:
if you are doing inband, when do you care whether the interface exists or not. by default all interface should be created by swss. in line 58, you have hotplug meaning you can wait for the interface to appear and then start doing dhcp.

Reply:
Agreed and I will remove the interface exists check. 

Also, the hotplug even though mentioned in the configuration, the feature is not fully functional in ifupdown2. I checked with the ifupdown2 maintainer and it is on the ToDo list.